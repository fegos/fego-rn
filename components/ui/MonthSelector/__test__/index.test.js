import React from 'react';
import Renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import MonthPicker from '../index';

describe('MonthPicker Tests', () => {
  let data;
  beforeEach(() => {
    data = [{
      label: 'item1',
      value: '1',
    }, {
      label: 'item2',
      value: '2',
    }, {
      label: 'item3',
      value: '3',
    }, {
      label: 'item4',
      value: '4',
    }, {
      label: 'item5',
      value: '5',
    }];
  });

  it('renders correctly', () => {
    const tree = Renderer.create(<MonthPicker data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('default initialValue tests', () => {
    const wrapper = shallow(<MonthPicker
      data={data}
    />);
    expect(wrapper.instance().state.value).toBe('5');
  });

  it('close monthPicker event', () => {
    const handler = jest.fn();
    const wrapper = shallow(<MonthPicker
      data={data}
      onClose={handler}
    />);
    // 找到取消按钮
    wrapper.childAt(1).simulate('press');
    expect(handler).toHaveBeenCalled();
  });

  it('select event', () => {
    const handler = jest.fn();
    const wrapper = shallow(<MonthPicker
      initialValue="3"
      data={data}
      onSelect={handler}
    />);
    // 初始选中3
    expect(wrapper.instance().state.value).toBe('3');
    // 找到2并选择
    wrapper.childAt(0).childAt(0).childAt(1)
      .simulate('press');
    expect(wrapper.instance().state.value).toBe('2');
  });
});

