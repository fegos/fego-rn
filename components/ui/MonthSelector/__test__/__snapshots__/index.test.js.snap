// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthPicker Tests renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F1F1F3",
        "justifyContent": "flex-start",
      },
      Object {
        "height": 254,
      },
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {},
        Object {
          "height": 200,
        },
      ]
    }
  >
    <RCTScrollView
      alwaysBounceHorizontal={false}
      alwaysBounceVertical={false}
      automaticallyAdjustContentInsets={false}
      bounce={false}
      onScrollEndDrag={[Function]}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
    >
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#F1F1F3",
                "borderBottomColor": "#ededed",
                "borderBottomWidth": 0.5,
                "height": 50,
                "marginLeft": 0,
                "paddingBottom": 16,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 16,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "color": "#858DA0",
                    "flex": 1,
                    "fontSize": 16,
                    "justifyContent": "center",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              item1
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#F1F1F3",
                "borderBottomColor": "#ededed",
                "borderBottomWidth": 0.5,
                "height": 50,
                "marginLeft": 0,
                "paddingBottom": 16,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 16,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "color": "#858DA0",
                    "flex": 1,
                    "fontSize": 16,
                    "justifyContent": "center",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              item2
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#F1F1F3",
                "borderBottomColor": "#ededed",
                "borderBottomWidth": 0.5,
                "height": 50,
                "marginLeft": 0,
                "paddingBottom": 16,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 16,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "color": "#858DA0",
                    "flex": 1,
                    "fontSize": 16,
                    "justifyContent": "center",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              item3
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#F1F1F3",
                "borderBottomColor": "#ededed",
                "borderBottomWidth": 0.5,
                "height": 50,
                "marginLeft": 0,
                "paddingBottom": 16,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 16,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "color": "#858DA0",
                    "flex": 1,
                    "fontSize": 16,
                    "justifyContent": "center",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              item4
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#F1F1F3",
                "borderBottomColor": "#ededed",
                "borderBottomWidth": 0.5,
                "height": 50,
                "marginLeft": 0,
                "paddingBottom": 16,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 16,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "color": "#858DA0",
                    "flex": 1,
                    "fontSize": 16,
                    "justifyContent": "center",
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#333333",
                  },
                ]
              }
            >
              item5
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderColor": "#999",
        "borderRadius": 3,
        "borderWidth": 0,
        "flexDirection": "row",
        "height": 44,
        "justifyContent": "center",
        "marginTop": 10,
        "opacity": 1,
        "paddingHorizontal": 13,
      }
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#858DA0",
              "fontSize": 16,
              "textAlign": "center",
            },
            Object {},
          ]
        }
      >
        取消
      </Text>
    </View>
  </View>
</View>
`;
